sdk/core/perf/inc/azure/perf/random_stream.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief A random stream of any specific size. Useful for test cases.
+ *
+ */
+
+#pragma once
+
+#include <azure/core/io/body_stream.hpp>
+
+#include <memory>
+#include <random>
+
+namespace Azure { namespace Perf {
+
+  /**
+   * @brief A random stream of any specific size. Useful for test cases.
+   *
+   */
+  class RandomStream {
```

Comment:
What is this? Why do we need it?

Reply:
RandomStream is an utility class offered by the Performance Framework.  

In summary, it can provide a BodyStream of any size by re-using a 1Mb buffer.  For example, I can ask for a 1TB stream and it is going to happen is that a 1Mb buffer is allocated and filled with random data. Then, it is used to read over and over from it until 1TB is read.

This is exposed in the other languages as a way to test the Upload API with different sizes, or any other API from a service that requires a bodyStream.

In other languages, Upload can do multi-thread from one single stream. Since that's not possible on C++, after creating the randomStream, I call ReadToEnd to allocate the entire memory and call UploadFrom(...) API so we can do multi-thread.  This will work as long as we don't have test of uploading more memory than the system available.

And this is useful to test the single Upload(..) API with any size we want.

Note: This is not being added to Storage. It is part of the Perf Framework